pvanalytics/features/clipping.py

```diff
@@ -88,3 +88,141 @@ def levels(ac_power, window=4, fraction_in_window=0.75,
         flags = flags | _label_clipping(temp, window=window,
                                         frac=fraction_in_window)
     return flags
+
+
+def _daytime_powercurve(ac_power, power_quantile=0.995,
+                        frequency_quantile=0.25):
+    # return the 99.5% quantile of daytime power data after removing
+    # night time data.
+    minutes = ac_power.index.hour * 60 + ac_power.index.minute
+    positive_power = ac_power >= 0
+    powerfreq = positive_power.groupby(minutes).sum()
+    min_daytime_freq = powerfreq.quantile(frequency_quantile)
+    daytimes = powerfreq[powerfreq >= min_daytime_freq].index
+    daytime_power = ac_power[minutes.isin(daytimes)]
+
+    return daytime_power.groupby(
+        daytime_power.index.hour * 60 + daytime_power.index.minute
+    ).quantile(power_quantile)
+
+
+def _clipped(power, derivative, power_min, derivative_max):
+    # return a mask that is true where `power` is greater than
+    # `power_min` and the absolute value of `derivative` is less
+    # than `derivative_max`
+    return (np.abs(derivative) <= derivative_max) & (power > power_min)
+
+
+def _clipping_power(ac_power, derivative_max=0.0035, power_min=0.75,
+                    power_quantile=0.995, frequency_quantile=0.25,
+                    freq=None):
+    # Calculate a power threshold, above which the power is being
+    # clipped.
+    #
+    # - The daytime power curve is calculated using
+    #   _daytime_powercurve(). This function groups `ac_power` by
+    #   minute of the day and returns the `power_quantile`-quantile of
+    #   power for each minute, excluding night time, early morning,
+    #   and late afternoon/evening (the amount of data that is
+    #   excluded is controlled by `frequency_quantile`).
```

Comment:
use `:py:func:` < full path to frequency_quantile > `

Reply:
This is a parameter, actually. We apply two quantiles: `power_quantile` which forms the power curve, and `frequency_quantile`.  I made a mistake here and forgot to pass it to `_daytime_powercurve`.

Reply:
Fixed this in ba688a5